Nodejs/Product/Nodejs/Project/NodejsProjectLauncher.cs

```diff
@@ -215,6 +217,12 @@ internal static string GetFullUrl(string host, int port)
             return builder.ToString();
         }
 
+        internal static bool ShouldUseV3CdpDebugger()
+        {
+            var featureFlagsService = (IVsFeatureFlags)ServiceProvider.GlobalProvider.GetService(typeof(SVsFeatureFlags));
+            return featureFlagsService != null && featureFlagsService.IsFeatureEnabled("JavaScript.Debugger.V3CdpDebugAdapter", false);
```

Comment:
I think it's fine to require a VS restart to pick up a feature flag change (I'm not even sure it _can_ change during a session), so I'd cache the result after retrieving the first time.  I might even compute it eagerly if that let me guarantee I was on the UI thread.

Reply:
Caching might get confusing though since the feature flag we're using (under Tools->Options->Environment->Preview Features) can change within a session of VS (though not within a debugging session in VS). I've added a check to make sure we're on the UI thead though! If we're not on the UI thread, it just doesn't launch the debugger and shows an error dialog about needing to be on the UI thread.